src/mcmcchains.jl

```diff
@@ -76,7 +76,16 @@ function section_dict(chns::Chains, section)
         sizes = reduce((a, b) -> max.(a, b), locs)
         ndim = length(sizes)
         var_views = (@view(chns.value[:, n, :]) for n in loc_names)
-        oldarr = reshape_values(replacemissing(convert(Array, cat(var_views...; dims=3))))
+        oldarr = let init = first(var_views)
+            # Splatting can be _very_ slow when `length(var_views)` is high,
+            # sometimes even resulting in `StackOverflowError`. Hence we use `reduce` with `cat` instead.
+            arr = reduce(
+                (x, y) -> cat(x, y; dims=3),
+                Iterators.drop(var_views, 1);
+                init=reshape(init, size(init)..., 1),
+            )
+            reshape_values(replacemissing(convert(Array, arr)))
+        end
```

Comment:
The existing code in this function was pretty hideous and hackish. Instead of working around it, I think we can both clean up the existing code, reduce allocations, and fix this bug by deleting `var_views` and then calling the following:
```suggestion
        # NOTE: slicing specific entries from AxisArrays does not preserve order
        # https://github.com/JuliaArrays/AxisArrays.jl/issues/182
        oldarr = permutedims(reshape_values(chns.value[:, loc_names, :]))
```

This keeps allocations low and code simple. Then, all we need to do is update lines 94-96 to be (throwing in a `@views` for good measure):
```julia
            for i in eachindex(locs, loc_names)
                @views arr[:, :, locs[i]...] = oldarr[:, :, loc_names[i]]
            end
```

And lastly, update `reshape_values` to use the permutation `(3, 1, 2)`.

What do you think?

Reply:
When fixing this I was also thinking "it's probably best to just set the values in a pre-allocated array using a `for` instead" but didn't want to make too many changes :sweat_smile: So yeah, I like that :+1:

Reply:
@sethaxen I decided to revert this change because I realized I unfortunately do not have the time to fix it right now :confused:

Reply:
No problem! Would you mind if I opened a PR to fix this that would supersede this fix?

Reply:
Feel free to commit to whatever you like:) I'm not using this branch locally or anything.